### ISSUE TITLE
(fedora) Other window manager detected

### ISSUE THREAD

**Issue by rrix** [ch11ng/exwm#31]
Hi again,

So latest master of all three projects got me to the point where Emacs would startup without timeout, but EXWM is still not coming up. Upon startup with the following xinit, I get "[EXWM] Other window manager detected"

```
xhost +
exec dbus-launch --exit-with-session emacs -l ~/.emacs.d/init.el
```

I created a stub xinit to see what exactly *was* shown by X11, and have:

```
xhost +
#exec dbus-launch --exit-with-session emacs -l ~/.emacs.d/init.el
xprop -root >~/xp
echo "AAAA" >> ~/xp
xwininfo -root -children -all >> ~/xp
```

and xp is as follows:

```
_XKB_RULES_NAMES(STRING) = "evdev", "pc105", "us", "", ""
XFree86_has_VT(INTEGER) = 1
XFree86_VT(INTEGER) = 3
AAAA

xwininfo: Window id: 0xd9 (the root window) (has no name)

  Root window id: 0xd9 (the root window) (has no name)
  Parent window id: 0x0 (none)
     0 children.

  Absolute upper-left X:  0
  Absolute upper-left Y:  0
  Relative upper-left X:  0
  Relative upper-left Y:  0
  Width: 1366
  Height: 768
  Depth: 24
  Visual: 0x20
  Visual Class: TrueColor
  Border width: 0
  Class: InputOutput
  Colormap: 0x22 (installed)
  Bit Gravity State: ForgetGravity
  Window Gravity State: NorthWestGravity
  Backing Store State: NotUseful
  Save Under State: no
  Map State: IsViewable
  Override Redirect State: no
  Corners:  +0+0  -0+0  -0-0  +0-0
  -geometry 1366x768+0+0

  Bit gravity: ForgetGravity
  Window gravity: NorthWestGravity
  Backing-store hint: NotUseful
  Backing-planes to be preserved: 0xffffffff
  Backing pixel: 0
  Save-unders: No

  Someone wants these events:
  Do not propagate these events:
  Override redirection?: No

  No window manager hints defined
  Window manager hints:
      Process id: (unknown)

  No normal window size hints defined
  No zoom window size hints defined

  No window shape defined
  No border shape defined

```

I tried modifying `exwm-init` to look for the `Window manager hints` to be defined, but was unsuccessful. It seems like simply making sure a root window is not set is not enough on Fedora?

**Comment by ch11ng** [comment]
Hi Ryan,
This is due to a bug in XCB and I'll seek a way to fix it (upstream or just in XELB).

**Comment by rrix** [comment]
Thanks for being so responsive :) This is a super neat project and I want to make it my daily driver ASAP :)

**Comment by ch11ng** [comment]
Hi Ryan,
This issue should have been fixed in ch11ng/xelb@aaddcd92ffd8a656127c457480c96185742b2d25.

**Comment by rrix** [comment]
And that made this bad buy work! Thanks so much for this work, I am super excited
